.github/workflows/changelog_check.yml

```diff
@@ -1,17 +0,0 @@
-name: Changelog
```

Comment:
is there a particular reason for coalescing all the pipelines into one? I quite liked the separation because it means there's much less GHA YAML to grok at once when you have to come and mess with it.

Reply:
Yes and no. Having stuff in one file (e.g. by using a `matrix` `strategy`) can help reduce duplication. I'm trying to prevent one file/project having an outdated bit of config where possible. I did deliberately lump the linting into one reusable action so that there's less yaml duplicated between projects.

No hard and fast rules here though.

Reply:
Oh, and AFAIK it's easiest to make one piece of CI conditional on another piece of CI when they're defined in the same file.